internal/git/git.go

```diff
@@ -2,11 +2,29 @@ package git
 
 import (
 	"fmt"
+	"path"
 	"strings"
 
 	"github.com/magefile/mage/sh"
 )
 
+// RepoNameFromURL returns the repository name
+func RepoNameFromURL() string {
+	url, err := RepoURL()
+
+	if err != nil {
+		return ""
+	}
```

Comment:
i think it might be better to return an error here, else further down the road it might end up with unwanted empty strings.